Automattic/wp-calypso
# HelpCenter: show sibyl articles before submitting a ticket or starting a chat

#### Proposed Changes

* This PR displays article results to the user before he is allowed to send an email or initiate a chat session
* This PR fixes how we handle sibyl articles without `post_id`

#### Testing Instructions

##### Setup
1. Checkout this branch
2. run 'yarn && yarn start'
3. cd into apps/editing-toolkit and run `yarn dev --sync`
4. cd into apps/happychat and run `yarn dev --sync`
5. Login to https://hud-staging.happychat.io/ and start accepting chats

##### Testing the chat
1. Open the help center
2. Request to chat
3. In the text area write `How Do I add a paypal`
4. The submit button will change to "Continue"
5. Click the submit button
6. You will see: back button, header 'These are some helpful articles', a list of articles, and a submit button 'Still chat with us'

##### Testing the chat article navigation
1. Click on one of the suggested articles
2. Click on the back button - you should see the previous page
3. Click the back button again - you should see the contact form with the 'Continue' button

##### Testing opening of a new chat
1. Click on 'Still chat with us'
2. The text you used initially is used to open a new chat session

##### Testing the chat article suggestion without `post_id`
1. Open the help center
2. Request to chat
3. In the text area write `How do I add a survey?`
4. The submit button will change to "Continue"
5. You should see something similar to this: 
<img width="418" alt="Screenshot 2022-10-04 at 18 20 39" src="https://user-images.githubusercontent.com/7000684/193872633-0ebc719a-24c6-4465-8fa2-e3511afe5a4d.png">
6. Clicking on the external article will open a new tab

##### Testing contact via email
1. Open the help center
2. Request to contact by email
3. Write something in the subject field
4. In the text area write `How Do I add a paypal`
5. The submit button will change to "Continue"
7. Click the submit button
8. You will see: back button, header 'These are some helpful articles', a list of articles, and a submit button 'Still email us'.
9. Click the submit button and ensure everything works properly

Related to https://github.com/Automattic/wp-calypso/issues/68569
